### PR TITLE
Adding icon and panel to Facebook Container

### LINKED ISSUE #203: Add browserAction and panel for after-install onboarding
As depicted in https://mozilla.invisionapp.com/share/52QH95R7CTV#/screens/348257325

First panel after install ...

image

Clicking "next" should go to the next panel ...

image

Note: icon is not final yet; can use a substitute icon for now.

### PR DESCRIPTION
**Pull request by ShivangiKakkar**
### Fixes #203

### CODE REVIEW

File: src/panel.css [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          line-height: 1.5em;
+          margin-top: 0.5em;
+          position: relative;
+          border-bottom:1px solid #dbd5d5;
+          zoom: 1;
+          padding-bottom:10px;
+}
+          
+.profile :after {
+              width: 100%;
```
Comment (jonathanKingston) [inline src/panel.css]: Check the alignment in this file as it's not consistent with itself. I suspect we should use 2 spaces like is most common in Mozilla code.

File: src/manifest.json [hunk 2]
```diff
@@ -17,9 +17,17 @@
...
+        "browser_style": true,
+        "default_icon": "fb-ban.png",
+        "default_title": "Block-facebook",
+        "default_popup": "panel.html"
+      },
+
     "homepage_url": "https://github.com/mozilla/contain-facebook",
 
     "permissions": [
+        "activeTab",
```
Comment (jonathanKingston) [inline src/manifest.json]: I'm not sure that we need this as we have `<all_urls>`

File: src/panel.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<html>
+    <head>
+      <title>Block-facebook</title>
+      <link rel="stylesheet" href="panel.css">
+    </head>
+
+    <body>
+      <div id="panel-1" class="wrapper">
```
Comment (jonathanKingston) [inline src/panel.html]: As mentioned split this into two files and set the html page in the browser.js file.

File: src/panel.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      <div>
+          <button id=2 class="button">Next</button>
+      </div>
+
+      <div id="panel-2" class="wrapper">
+          <h3 class="profile">
+                Facebook Container
+          </h3>
+          <p>
+            On your next visit to Facebook it will be opened in a container, which will isolate your Facebook identy from other websites.
```
Comment (jonathanKingston) [inline src/panel.html]: Check the alignment in this file, I think everywhere else we use 2 spaces indenting.

File: src/panel.css [hunk 5]
```diff
@@ -0,0 +1,41 @@
+.wrapper {
+    width: 280px;
+    height: 170px;
+    padding:10px 10px 10px 10px;
```
Comment (jonathanKingston) [inline src/panel.css]: Ensure there is always a space after `padding:` and other properties.

**Comment by ShivangiKakkar** [comment]
It somewhat looks like: (the icons are pseudo right know)

screenshot 1940-12-15 at 2 03 48 am

File: src/panel.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    <title>Block-facebook</title>
+    <link rel="stylesheet" href="panel.css">
+  </head>
+  <body>
+    <div id="panel-1" class="wrapper">
+      <h4 class="profile heading padding">
+        Facebook Container
+      </h4>
+      <p class="padding">
+        Facebook Container has been installed and is preventing Facebook from tracking you around the web. Anytime you see <img src="fb-ban2.png"> this addon is blocking Facebook from traking you.
```
Comment (pdehaan) [inline src/panel.html]: > from traking you

typo: "tracking"

File: src/panel2.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    <link rel="stylesheet" href="panel.css">
+  </head>
+
+  <body>
+    <div id="panel-2" class="wrapper">
+      <h4 class="profile heading padding">
+        Facebook Container
+      </h4>
+      <p class="padding">
+        On your next visit to Facebook it will be opened in a container, which will isolate your Facebook identy from other websites.
```
Comment (pdehaan) [inline src/panel2.html]: > identy

typo: "identity"

File: src/panel.css [hunk 8]
```diff
@@ -0,0 +1,30 @@
+.wrapper {
+  width: 280px;
+  height: 170px;
+  box-shadow: 0 8px 40px rgba(#000000, .2);
```
Comment (pdehaan) [inline src/panel.css]: Does CSS support hex for RGBA colors, or is that just a Sass thing?
Should this be rewritten as `rgba(0, 0, 0, .2)`?

File: src/panel.html (hunk 6 above)
Comment (pdehaan) [inline src/panel.html]: Also, should it be "add-on" (which matches the README.md usage).

Comment (ShivangiKakkar) [reply src/panel.html]: I am so sorry for typos I am correcting them. I just typed them in hurry.

File: src/panel.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=utf-8">
+    <title>Block-facebook</title>
```
Comment (pdehaan) [inline src/panel.html]: Should this be "Block-facebook" or "Facebook Container" (although possibly irrelevant if the title isn't visible anywhere)

File: src/panel.html [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    <link rel="stylesheet" href="panel.css">
+  </head>
+  <body>
+    <div id="panel-1" class="wrapper">
+      <h4 class="profile heading padding">
+        Facebook Container
+      </h4>
+      <p class="padding">
+        Facebook Container has been installed and is preventing Facebook from tracking you around the web. Anytime you see <img src="fb-ban2.png"> this add-on is blocking Facebook from tracking you.
```
Comment (pdehaan) [inline src/panel.html]: Also, are these the correct icons? I tried verifying against the Invision screens in #203, but I think I counted at least 5 different toolbar icons for Facebook Container in that deck, so I wasn't sure what the final ones were.

File: src/panel.html [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+      </p>  
+    </div>
+    <div>
+      <button class="button">Next</button>
+    </div>
+
+    <script src="panel.js"></script>
```
Comment (pdehaan) [inline src/panel.html]: I was a bit confused here, since the panel.js file looks empty in the diff below.

File: src/panel2.html [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+  <body>
+    <div id="panel-2" class="wrapper">
+      <h4 class="profile heading padding">
+        Facebook Container
+      </h4>
+      <p class="padding">
+        On your next visit to Facebook it will be opened in a container, which will isolate your Facebook identity from other websites.
+      </p>
+      <p class="text padding">Visit Facebook</p>
```
Comment (pdehaan) [inline src/panel2.html]: Should this be a link?

File: src/manifest.json [hunk 13]
```diff
@@ -17,6 +17,13 @@
       }
     },
 
+    "browser_action": {
+        "browser_style": true,
+        "default_icon": "fb-ban.png",
+        "default_title": "Block-facebook",
```
Comment (pdehaan) [inline src/manifest.json]: Not sure where/if these titles are user visible, but should this be "Facebook Container"?

File: src/panel.html (hunk 10 above)
Comment (ShivangiKakkar) [reply src/panel.html]: These are test icons only. The final one needs a platform level change.

**Comment by ShivangiKakkar** [comment]
@groovecoder The icon is appearing right after we run. How do I make it not appear first and appear after we install facebook container?

>Fix the "Visit Facebook" link

If I add it as hyperlink it opens the fb site twice.
> Fix the "Got it" button so it closes the browserAction panel

It is showing a small empty popup couldn't close it completely.

>Does that sound right? Do we know when we might have the final icons?

I guess the fence icon may be. (it was assigned to me on bugzilla to change in platform level)

**Comment by ShivangiKakkar** [comment]
@groovecoder I tried to squash all commits, I guess I messed it all.

**Comment by groovecoder** [comment]
Superseded by https://github.com/mozilla/contain-facebook/pull/214

**Pull request closed without merging**